integration/v3_grpc_test.go

```diff
@@ -1006,3 +1007,97 @@ func TestV3RangeRequest(t *testing.T) {
 		clus.Terminate(t)
 	}
 }
+
+// TestV3LeaseRevoke ensures a key is deleted once its lease is revoked.
+func TestV3LeaseRevoke(t *testing.T) {
+	testLeaseAction(t, func(lc pb.LeaseClient, leaseID int64) error {
+		_, err := lc.LeaseRevoke(context.TODO(), &pb.LeaseRevokeRequest{ID: leaseID})
+		return err
+	})
+}
+
+// TestV3LeaseCreateById ensures leases may be created by a given id.
+func TestV3LeaseCreateByID(t *testing.T) {
+	clus := newClusterGRPC(t, &clusterConfig{size: 3})
+	defer clus.Terminate(t)
+
+	// create fixed lease
+	lresp, err := pb.NewLeaseClient(clus.RandConn()).LeaseCreate(
+		context.TODO(),
+		&pb.LeaseCreateRequest{ID: 1, TTL: 1})
+	if err != nil {
+		t.Errorf("could not create lease 1 (%v)", err)
+	}
+	if lresp.ID != 1 {
+		t.Errorf("got id %v, wanted id %v", lresp.ID)
+	}
+
+	// create duplicate fixed lease
+	lresp, err = pb.NewLeaseClient(clus.RandConn()).LeaseCreate(
+		context.TODO(),
+		&pb.LeaseCreateRequest{ID: 1, TTL: 1})
+	if err != nil {
+		t.Error(err)
+	}
+	if lresp.ID != 0 || lresp.Error != lease.ErrLeaseExists.Error() {
+		t.Errorf("got id %v, wanted id 0 (%v)", lresp.ID, lresp.Error)
+	}
+
+	// create fresh fixed lease
+	lresp, err = pb.NewLeaseClient(clus.RandConn()).LeaseCreate(
+		context.TODO(),
+		&pb.LeaseCreateRequest{ID: 2, TTL: 1})
+	if err != nil {
+		t.Errorf("could not create lease 2 (%v)", err)
+	}
+	if lresp.ID != 2 {
+		t.Errorf("got id %v, wanted id %v", lresp.ID)
+	}
+
+}
+
+// acquireLeaseAndKey creates a new lease and creates an attached key.
+func acquireLeaseAndKey(clus *clusterV3, key string) (int64, error) {
+	// create lease
+	lresp, err := pb.NewLeaseClient(clus.RandConn()).LeaseCreate(
+		context.TODO(),
+		&pb.LeaseCreateRequest{TTL: 1})
+	if err != nil {
+		return 0, err
+	}
+	if lresp.Error != "" {
+		return 0, fmt.Errorf(lresp.Error)
+	}
+	// attach to key
+	put := &pb.PutRequest{Key: []byte(key), Lease: lresp.ID}
+	if _, err := pb.NewKVClient(clus.RandConn()).Put(context.TODO(), put); err != nil {
+		return 0, err
+	}
+	return lresp.ID, nil
+}
+
+// testLeaseAction performs some action while holding a lease with an
+// attached key, then confirms the key is gone.
+func testLeaseAction(t *testing.T, act func(pb.LeaseClient, int64) error) {
```

Comment:
I feel this func name is not descriptive. The functionality of this is to create a leased key and then confirm leased key is removed. `testLeaseAction` sounds like we test the lease action itself.